tools/find_image.rb

```diff
@@ -33,14 +33,14 @@ def run
           :url => "https://openshift-release.svc.ci.openshift.org/"
         raise "missing query string" if options.query.nil?
         print("Querying #{options.url} against #{options.build_type}\n")
-        doc = Nokogiri::HTML(open(options.url).read)
+        doc = Nokogiri::HTML(RestClient.get(options.url).to_s)
```

Comment:
Did you try `URI.open` instead of `RestClient`.

Reply:
@akostadinov I don't think that suggested text is correct as URI does not have a public method that's named `open` https://ruby-doc.org/stdlib-2.5.1/libdoc/uri/rdoc/URI.html

Reply:
`require 'open-uri'` makes this method available. Try it.

Reply:
@akostadinov that worked, but it's requires more work than simply using the lib `rest-client` which is part of the [Gemfile](https://github.com/openshift/verification-tests/blob/master/Gemfile#L5).  For example, the following is equivalent of `RestClient.get(options.url).to_s`

```addr = "https://openshift-release.svc.ci.openshift.org/"
uri = URI.parse(addr)
tries = 3
begin
  res = uri.open(redirect: false)
rescue OpenURI::HTTPRedirect => redirect
  binding.pry
  uri = redirect.uri # assigned from the "Location" response header
  retry if (tries -= 1) > 0
  raise
end
doc = open(res).read
```

Reply:
Have you tried that redirection doesn't work with `URI.open(addr)`? For me redirections worked. Not sure I tried exactly your URL.

Reply:
@akostadinov, nope, that didn't work ```[1] pry(main)> require 'open-uri'
=> true
[2] pry(main)> URI.open('https://openshift-release.svc.ci.openshift.org/')
RuntimeError: redirection forbidden: https://openshift-release.svc.ci.openshift.org/ -> http://openshift-release.apps.ci.l2s4.p1.openshiftapps.com/
from /Users/pruan/.rvm/rubies/ruby-2.6.6/lib/ruby/2.6.0/open-uri.rb:239:in `open_loop'```

Reply:
This is actually a security issue on the registry side. HTTPS redirects to a HTTP URL. So user can't trust that what they get is legitimate. It strips off HTTPS protection. We need to report this to the URL administrator.

Reply:
`URI.open` does support redirection between HTTPS URLs. As well from HTTP->HTTPS since Ruby 2.4. But it correctly in my opinion refuses to redirect back to HTTP.

Reply:
the final re-direct does point to https://openshift-release.apps.ci.l2s4.p1.openshiftapps.com/. so I'm wondering if that `URI.open` can ignore the intermediate `http` url.  Anyway, what's the objection of using `RestClient.get` to get around the issue while whatever team is responsible notified and the fix is implemented and merged?